modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java

```diff
@@ -1060,7 +1074,7 @@ else if (affinityCache.version().compareTo(resAffinityVer) > 0) {
      * @throws SQLException On any error.
      */
     void sendQueryCancelRequest(JdbcQueryCancelRequest req, JdbcThinTcpIo cliIo) throws SQLException {
-        if (!connected)
+        if (connCnt.get() == 0)
```

Comment:
Also I wonder why we check the counter to determine if we have working connections or not? Cannot we use `ios.isEmpty()` instead?

Reply:
Nope, after some changes inside onDisconnect(JdbcThinTcpIo cliIo) we might use something like  `if (affinityAwareness && ios.isEmpty() || !affinityAwareness && singleIo == null)`, however `connCnt.get() == 0 `is faster.